german-quinonez/lesson-1/homework-2-Car/src/Car.java

```diff
@@ -0,0 +1,65 @@
+public class Car {
+
+    private double gas;
+    private double distance;
+    private double gasConsume;
+    private boolean isPowerOn;
+    private double capacity;
+
+
+    public Car() {
+        this.gas = 0.0;
+        this.distance = 0.0;
+        this.gasConsume = 0.0;
+        this.isPowerOn = false;
+        this.capacity = 0.0;
+    }
+
+    public void powerOn() {
+        this.isPowerOn = true;
+    }
+
+    public void powerOff() {
+        this.isPowerOn = false;
+    }
+
+    public void loadGas(double gas) {
+        if (this.gas + gas <= this.capacity) {
+            this.gas += gas;
+        } else {
+            this.gas = capacity;
+        }
+    }
+
+    public void move(double distance) {
+        double maxDistance = gasConsume * gas;
+        if (distance <= maxDistance) {
+            this.distance += distance;
+        } else {
+            this.distance += maxDistance;
+        }
+        this.gas -= this.distance / this.gasConsume;
```

Comment:
In case that maxDistance is less than distance gas will less than 0 with this approach, so make gas 0 when distance  is greater than maxDistance